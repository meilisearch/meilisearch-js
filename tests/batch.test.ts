import { afterAll, beforeEach, describe, expect, test } from "vitest";
import {
  config,
  getClient,
  clearAllIndexes,
} from "./utils/meilisearch-test-utils.js";

const index = { uid: "batch-test" };

afterAll(() => {
  return clearAllIndexes(config);
});

describe.each([{ permission: "Master" }, { permission: "Admin" }])(
  "Tests on batches",
  ({ permission }) => {
    beforeEach(async () => {
      const client = await getClient("Master");
      await client.createIndex(index).waitTask();
    });

    test(`${permission} key: Get all batches`, async () => {
      const client = await getClient(permission);
      const batches = await client.batches.getBatches();
      expect(batches.results.length).toBeGreaterThan(0);
    });

    test(`${permission} key: Get one batch`, async () => {
      const client = await getClient(permission);

      const batches = await client.batches.getBatches();
      const batch = await client.batches.getBatch(batches.results[0].uid);
      expect(batch.uid).toEqual(batches.results[0].uid);

      // Can't use toMatchSnapshot because the output changes every time
      expect(batch.details).toBeDefined();
      expect(batch.stats).toHaveProperty("totalNbTasks");
      expect(batch.stats).toHaveProperty("status");
      expect(batch.stats).toHaveProperty("types");
      expect(batch.stats).toHaveProperty("indexUids");
      expect(batch.stats).toHaveProperty("progressTrace");
      expect(batch.duration).toBeDefined();
      expect(batch.startedAt).toBeDefined();
      expect(batch.finishedAt).toBeDefined();
      expect(batch.progress).toBeDefined();
    });
  },
);
