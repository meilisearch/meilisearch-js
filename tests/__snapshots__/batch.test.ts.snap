// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tests on batches > Admin key: Get one batch 1`] = `
Batch {
  "details": {},
  "duration": "PT0.000219583S",
  "finishedAt": "2025-04-01T07:00:51.408688632Z",
  "progress": null,
  "startedAt": "2025-04-01T07:00:51.408469049Z",
  "stats": {
    "indexUids": {
      "batch-test": 1,
    },
    "progressTrace": {
      "processing tasks": "180.17µs",
      "processing tasks > creating the index": "93.63µs",
      "writing tasks to disk": "64.29µs",
      "writing tasks to disk > task": "47.54µs",
    },
    "status": {
      "failed": 1,
    },
    "totalNbTasks": 1,
    "types": {
      "indexCreation": 1,
    },
  },
  "uid": 1839,
}
`;

exports[`Tests on batches > Master key: Get one batch 1`] = `
Batch {
  "details": {},
  "duration": "PT0.000211542S",
  "finishedAt": "2025-04-01T07:00:51.336459757Z",
  "progress": null,
  "startedAt": "2025-04-01T07:00:51.336248215Z",
  "stats": {
    "indexUids": {
      "batch-test": 1,
    },
    "progressTrace": {
      "processing tasks": "164.50µs",
      "processing tasks > creating the index": "82.04µs",
      "writing tasks to disk": "80.25µs",
      "writing tasks to disk > task": "63.92µs",
    },
    "status": {
      "failed": 1,
    },
    "totalNbTasks": 1,
    "types": {
      "indexCreation": 1,
    },
  },
  "uid": 1837,
}
`;
